### PR TITLE
WM-2303, WM-2341: Workflows user metrics

### PR DESCRIPTION
**Pull request by cjllanwarne**
What:
- Updates EventWorkspaceDetails to include a workspace role (OWNER, WRITER, READER)
- Updates workflows app import buttons to emit an imported event.

Why:
- Measuring usage and success of multi-user workflows

### Visual Aids
Example of a new `workflowsApp:import` event, including the new workspaceAccessLevel field:
image

### CODE REVIEW

File: src/workflows-app/FeaturedWorkflows.ts [hunk 1]
```diff
@@ -129,21 +131,30 @@ export const FeaturedWorkflows = ({
             const {
               cbasProxyUrlState: { state },
             } = await loadAppUrls(workspaceId, 'cbasProxyUrlState');
+            const importMethod = (state, method) => {
```
Comment (cjllanwarne) [inline src/workflows-app/FeaturedWorkflows.ts:134]: This felt odd adding a nested function, but putting it anywhere else I was running into static analysis errors about the `useCallback` closure... would love any feedback on improving this (or reassurance that it's fine as it is?)

File: src/libs/events.ts [hunk 2]
```diff
@@ -108,6 +113,7 @@ const eventsList = {
   workflowRerun: 'workflow:rerun',
   workflowUploadIO: 'workflow:uploadIO',
   workflowUseDefaultOutputs: 'workflow:useDefaultOutputs',
+  workflowsAppImport: 'workflowsApp:import',
```
Comment (dillydally414) [inline src/libs/events.ts:121]: Should this event also be fired when importing from Dockstore?

Comment (cjllanwarne) [reply src/libs/events.ts:121]: Do you mean when we use the featured workflows page? There's not really a difference between github and dockstore import from this page's point of view

Comment (dillydally414) [reply src/libs/events.ts:121]: I mean on this page where users land when importing a workflow from Dockstore https://github.com/DataBiosphere/terra-ui/blob/37a1975d9827fe0e6b9bcc2a3abe4e46c9923cd2/src/pages/ImportWorkflow/ImportWorkflow.js#L71

Comment (cjllanwarne) [reply src/libs/events.ts:121]: Good call, now added

**Review by dillydally414 — APPROVED** [review]

**Review by salonishah11 — APPROVED** [review]

**Pull request merged**